### PR TITLE
[GEOT-7225] Assertion in ShpFilesTestStream depends on environment

### PR DESCRIPTION
**Pull request by roarbra**
[GEOT-7225](https://osgeo-org.atlassian.net/browse/GEOT-7225) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

The fails on JDK 11, because of the assertion that isLocal will return false. For JDK 8 and earlier that was true, because the url would be jar:file... That has changed and the url will now be file:/.......jar!shapes/statepop.shp.

I don't see the need for that check, so I'm removing it.

I'm also renaming the file, so that the suffix of the class will be Test.

# Checklist

- [X] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [X] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [X] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [X] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [X] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [X] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [X] Bug fixes and small new features are presented as a single commit.
- [X] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by aaime** [comment]
I don't think it's JDK related, but more IDE related: maven will run the tests against the jars, while IDEs often do that against unpacked resources folders. The test is failing if I run it with JDK 8 from the IDE too.

**Comment by roarbra** [comment]
Yes, it might be the IDE as well.
I've installed it with a fresh JDK and Eclipse, so there could be one or the other.

File: modules/plugin/shapefile/src/test/java/org/geotools/data/shapefile/ShpFilesStreamTest.java [hunk 1]
```diff
@@ -182,13 +182,10 @@ public void testGetReadChannelFileChannel() throws IOException {
     }
 
     @Test
-    @SuppressWarnings("PMD.UnusedVariable") // really want to just open and close
```
Comment (aaime) [inline modules/plugin/shapefile/src/test/java/org/geotools/data/shapefile/ShpFilesStreamTest.java:185]: Oh interesting, PMD is no longer complaining that the "read" variable is not used? Nice find. Did you just try or have a way to find unnecessary PMD annotations?

**Review by aaime — COMMENTED** [review]
I'm good with removing the assert. Interested about the annotation removal, see below.

File: modules/plugin/shapefile/src/test/java/org/geotools/data/shapefile/ShpFilesStreamTest.java (hunk 1 above)
Comment (roarbra) [reply modules/plugin/shapefile/src/test/java/org/geotools/data/shapefile/ShpFilesStreamTest.java:185]: Just trying. Seemed wrong to say the variable was unused.

**Pull request merged**